src/index.py

```diff
@@ -1,44 +1,69 @@
 import sys
-from game.exception import MoveException
-from game.game import build_grid, movable_tiles, move, is_grid_resolved, shuffle
-from renderer.renderer import welcome, goodbye, shuffling, victory, show_action_not_valid, show_grid, ask_move, show_moves
+import os
+from game.exception import GridSizeNotValidException, MoveException
+from game.game import (
+    build_grid, movable_tiles, move, is_grid_resolved, shuffle
+)
+from renderer.renderer import (
+    welcome, goodbye, shuffling, victory, show_action_not_valid,
+    show_grid, show_moves, show_menu_size, show_size_not_valid,
+    ask_move, ask_size
+)
 
 
-def router(action, grid, shuffled):
+def router(action, grid, started_grid, shuffled):
     if not shuffled and action in ['s', 'S']:
         print(shuffling())
-        return [shuffle(grid), True]
+        return [shuffle(grid), started_grid, True]
+    elif action in ['w', 'W']:
+        new_grid = change_size()
+        return [new_grid, new_grid.copy(), False]
```

Comment:
Not mandatory for this PR but in this kind of function, we prefer return an object with named attributes.